(TK-298) Bring tk-status docs up to date

vilmibm opened the pull request:
This PR:
- Updates the documentation index
- Adds information on the simple endpoints
- Links to the status proxy docs

vilmibm committed:
(TK-298) Improve docs

This commit:

* Updates the documentation index
* Adds information on the simple endpoints
* Links to the status proxy docs

vilmibm committed:
(maint) clarify a test

In writing docs I wanted to ensure that :unknown still overrode
:starting and :stopping so I tweaked the relevant test.

KevinCorcoran commented on documentation/developers.md:
````diff
@@ -56,6 +55,31 @@ might be implemented to utilize the `compare-levels` function:
+}
+```
+
+For information on HTTP serving of the status endpoint, see the [Status Proxy documentation](./status-proxy-service.md).
````
I think this could probably be reworded to be a little clearer.  As it stands, my first thought is "well, I could just change the jetty configuration above to accept plaintext connections".  So maybe instead of "HTTP serving of the status endpoint" something like "enabling proxying of plaintext HTTP `/status` requests to an otherwise HTTPS-protected server" (or something less awkward).

KevinCorcoran commented on documentation/wire-formats.md:
```diff
@@ -66,7 +68,37 @@ of error that occurred.  For example, "service-status-version-not-found".
+
+#### Encoding
+
+The content type for these endpoints is `text/plain; charset=utf-8`.
```
This is quite redundant with the file above.

vilmibm replied:
Agreed, but I was following the established form in both files. Doesn't seem too bad to have it in both places

rlinehan commented on documentation/query-api.md:
```diff
@@ -212,3 +214,65 @@ version and a specific detail level:
+These endpoints are designed for load balancers that don't support any kind of
+JSON parsing or query parameter use. They return simple string bodies (either
+the state of the service in question or a simple error message) and a status
+code relevant to the status result.
```
I think it might be useful/help in clarifying if you say something like if you need non-https for your load balancer, you should use the proxy service with this.

vilmibm replied:
Updated.

vilmibm committed:
(TK-298) Clear up some things around HTTPS and proxying

It became clear that ever referring to the simple endpoints as being
"plaintext" was deeply confusing, so this commit rmoves such references.

KevinCorcoran commented:
👍 

@rlinehan any more feedback on this?

rlinehan commented:
Nope, lgtm, :shipit: .
